src/lib/server.ts

```diff
@@ -401,13 +403,16 @@ export class ZwavejsServer extends EventEmitter {
     this.sockets = new ClientsController(this.driver, this.logger);
     this.wsServer.on("connection", (socket) => this.sockets!.addSocket(socket));
 
-    this.logger.debug(`Starting server on port ${this.options.port}`);
+    const localEndpoint =
+      (this.options.host || "0.0.0.0") + ":" + this.options.port;
```

Comment:
If we also define a default here, let's not add one in `src/bin/server.ts` or we should at least extract this as a constant to ensure it will stay in sync.

Reply:
Hi @balloob,

The `localEndpoint` here was just to feed it to the log, it isn't manipulating the socket.
I used `0.0.0.0` as the http class will use that anyway if no host is provided.

I could remove this constant, and place it inline of the log entry if you prefer?

```javascript
/*
 * const localEndpoint =
 * (this.options.host || "0.0.0.0") + ":" + this.options.port;
 */

this.logger.debug(`Starting server on ${(this.options.host || "0.0.0.0") + ":" + this.options.port}`);

this.server.on("error", this.onError.bind(this));
this.server.listen(this.options.port, this.options.host);
await once(this.server, "listening");
this.emit("listening");
this.logger.info(`ZwaveJS server listening on ${(this.options.host || "0.0.0.0") + ":" + this.options.port}`);
```

As this is the final few lines before the socket is opened `host` and `port` will represent the users choice or any defaults used i.e those used in `bin/server.ts `if not cli params are provided.

Reply:
In that case src/bin/server shouldn’t apply defaults but we should do it just here.

Reply:
(For all values)

Reply:
Ok, that would make sense, `src/bin/server.bin` uses 3000 as its default port - Im guessing we should keep that.

Reply:
I think we can move that into the class too ?

Reply:
Yeah, sorry that's what meant, using 3000 as the default port in the main class.

Reply:
Ok @balloob @raman325 .

Defaults  have now been moved to the main class, here is what has changed.

 - `ZwavejsServerOptions` is now optional (given all its properties are now optional)
 - There is now a default port of `3000` and default host of `0.0.0.0` and this is applied on the main class.
 - Defaults have been moved away from **src/bin/server.ts**
 - I don't do much typescript 😉 
 
 Defaults are defined at class level.
```
private defaultPort: number = 3000;
private defaultHost: string = "0.0.0.0";
```